samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java

```diff
@@ -2974,12 +2974,12 @@ public int autoSubmitAssessments() {
 						" a.finalScore, a.comments, a.status, a.gradedBy, a.gradedDate, a.attemptDate, a.timeElapsed) " +
 						" from AssessmentGradingData a, PublishedAccessControl c " +
 						" where a.publishedAssessmentId = c.assessment.publishedAssessmentId " +
-						" and c.retractDate <= :retractDate" +
+						" and ((c.lateHandling = 1 and c.retractDate <= :currentTime) or c.dueDate <= :currentTime)" +
```

Comment:
So if due date has passed but retract date has not passed, this will now run?

Reply:
You're right, there is a logic error, the intention is to only respect the retract date if it is enabled, but this will check the due date if the retract date is not passed.
I'll update it to check the lateHandling value in both cases.

Reply:
We're all very skeptical of changes to this code right now..... a lot has been changed with the new, per-student exceptions ..... I'm going to be very skeptical of merging this code without a lot of testing .....